actions/actions.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/local/sbin/charm-env python3
+import os
+import pwd
+import grp
+import sys
+import tarfile
+import shutil
+
+from datetime import datetime
+from typing import List
+
+from charms import layer
+from charms.reactive.relations import endpoint_from_name
+from charmhelpers.core import hookenv
+from charmhelpers.core.hookenv import (
+    function_get,
+    function_set,
+    function_fail,
+    local_unit,
+    leader_set,
+    leader_get,
+)
+
+from reactive.easyrsa import (
+    easyrsa_directory,
+)
+
+# Import charm layers and start reactive
+layer.import_layer_libs()
+hookenv._run_atstart()
+
+PKI_BACKUP = '/home/ubuntu/easyrsa_backup'
+
+
+def backup() -> None:
+    uid = pwd.getpwnam("ubuntu").pw_uid
+    gid = grp.getgrnam("ubuntu").gr_gid
+    try:
+        os.mkdir(PKI_BACKUP, mode=0o700)
+    except FileExistsError:
+        pass
+
+    os.chown(PKI_BACKUP, uid, gid)
+    if not os.path.isdir(PKI_BACKUP):
+        raise RuntimeError('Backup destination is not a directory.')
+
+    timestamp = datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+    backup_name = 'easy-rsa-{}.tar.gz'.format(timestamp)
+    backup_path = os.path.join(PKI_BACKUP, backup_name)
+    with tarfile.open(backup_path, mode='w:gz') as tar:
+        tar.add(os.path.join(easyrsa_directory, 'pki'), 'pki')
+
+    function_set({
+        'command': 'juju scp {}:{} .'.format(local_unit(), backup_path),
+        'message': 'Backup archive created successfully. Use the juju scp '
+                   'command to copy it to your local machine.'
+    })
+
+
+def restore() -> None:
```

Comment:
Coud comments be added inline to give context more quickly on the steps followed to restore the data (not only the CA? what other PKI pieces are restored locally? which pieces are shared over the client relation?